### ISSUE TITLE
[Requesting Feedback] Update Schedule

### ISSUE THREAD

**Issue by ripose-jp** [ripose-jp/Memento#122]
I wanted to get feedback from users regarding when the next update should be.

At the time of writing, the last release was in June a little over 4 months ago. There have been some minor updates to functionality since then, but nothing too major. I think that v0.5.6 is the best release yet in terms of stability. Even the little bugs I've found so far aren't enough to warrant a new revision because I believe that most users will never notice them.

My intention has been to time the next release of Memento with the release of mpv [v0.35.0](https://github.com/mpv-player/mpv/issues/10662). This will add the ability to show secondary and primary subtitles at the same time. This has been something a lot of people have wanted for a long time. That said, timing Memento releases with every new mpv release probably isn't sustainable since mpv has settled into a cadence of releasing about once every year.

The second consideration I have with new releases is what to do about Qt6. Right now Memento uses Qt5 for everything which hasn't been much of a problem. Qt 5.15 is supported in every major Linux distro including even Ubuntu/Debian at this point. The issue is that The Qt Company has closed sourced their latest Qt5 releases for almost two years now in order to force people over to Qt6. Yet today still much of Linux is missing the critical infrastructure needed for Qt6 such as availability in repos, theme support from Breeze, and general support from KDE Plasma. This isn't something that affects Windows and macOS users, but it's a problem that's close to my heart. A version of Memento that compiles on Qt6 is available on the [qt6 branch](https://github.com/ripose-jp/Memento/tree/qt6), but I don't recommend using it since it be will seldom tested and regularly rebased on master.

I've been apprehensive to inundate users with updates that don't affect their user experience all that much. Right now there are several feature requests that have been implemented that still aren't widely available since most people just use the provided binaries. 

To get to the point, I want to know what sort of update schedule actual users want.

**Comment by Juiin** [comment]
For me personally, even small changes can make a big difference with a tool like this. I was pleasantly surprised with the small extra features i found, when building myself yesterday. So i think releasing, as soon as an update is somewhat stable would be ideal for me. But i cannot speak on the Linux issues, since i am a Windows user.

**Comment by pattontj** [comment]
I personally see no issues with a minor patch focused on stability or small feature improvements, and even if nobody is raising issues about bugs or or other things, there may still be people affected by it that have found ways of working around it. As for the Qt stuff, I don't see the immediate benefit(s) to updating to Qt6 from a user's perspective unless Qt5 is causing undue issues.

**Comment by dkaknbk** [comment]
I have also been very satisfied with the current stable release, I haven't noticed any bugs or missing features lately. I also wouldn't mind more frequent updates even for minor stuff, I use this program almost every day so even smaller QoL improvement features once in a while will probably have a positive impact on my user experience. I'm primarily a Windows user so I have no useful input about the Qt5/Qt6 issue, but I thought I'd mention that I have been using the Appimage version of Memento on the Steam Deck a bit and that seems to be running fine as well. The only minor QoL feature I could think off is adding commands related to subtitle delay into the Subtitle menu so they can be controlled via mouse so I don't have to connect a keyboard if I happen to have unsynced subtitles.

**Comment by ripose-jp** [comment]
>I thought I'd mention that I have been using the Appimage version of Memento on the Steam Deck a bit and that seems to be running fine as well.

This is a cool use case that I hadn't even imagined. I don't have any problem adding some features to the UI that are only accessible via keybinds for these sorts of devices.

On a side note, what I say about Qt doesn't apply to AppImages since those contain all the libraries they need except maybe glibc and libstdc++. Furthermore they don't theme well regardless of what Qt version they ship with, so they aren't even a consideration in this case. The fact that they don't theme and don't dynamically link to mpv are actually the only two reasons that I don't recommend using the AppImages. This is a personal gripe, but I find it frustrating that you're running Memento on a Steam Deck presumably running Plasma, which means Memento could look the way I intend (identical to the image on the front page), but instead probably looks like this:

![image](https://user-images.githubusercontent.com/72582120/195753891-a299e103-fcef-4937-981f-524e8a2b8629.png)

It's just that compiling Memento is such a high barrier to entry that there's no way I could ever stop supplying AppImages at this point. I had looked into FlatPak in the past which claims to solve this issue, but last time I looked into it they didn't have MeCab in their repositories. This is why I'm a big fan of the AUR. It solves this problem by reducing the compilation process to one command. Unfortunately only Arch-based distros seem to have this open repo model.
